### ISSUE TITLE
Line break, bold, italic, etc... inside labels

### ISSUE THREAD

**Issue by RaphaelAlvez** [ArnoldSmith86/virtualtabletop#36]
Probably using for example text markers like *this would be in bold*

**Comment by ArnoldSmith86** [comment]
Line break should work with `\n` in the JSON? I think? Maybe only add bold? I don't think we should overdo this. Although having text as text in the VTT (as opposed to images) makes it easier to translate games.

**Comment by larkob** [comment]
How about including a minimal markdown parser like https://github.com/developit/snarkdown

**Comment by ArnoldSmith86** [comment]
I mean: Really?! :'D

**Comment by larkob** [comment]
The code is already there - why not reuse it instead of doing it from scratch? If we implement "bold", sooner or later someone will ask for other features as well.

**Comment by ArnoldSmith86** [comment]
I mean, sure. But that's adding 90% bloat. I won't block this.

**Comment by larkob** [comment]
Yeah, actually, I was considering this just a feature request for "whenever", rather than an issue that needs to be addressed any time soon. Perhaps this should be moved somewhere where we can let people vote to see how popular it is.

**Comment by UltimateGeek** [comment]
snarkdown has about 20,000 weekly downloads.

There are a couple of populsar markdown renderers available as node modules:
* [markdown-it](https://github.com/markdown-it/markdown-it) - 1,000,000 weekly downloads
* [marked](https://github.com/markedjs/marked) - 2,400,000 weekly downloads

snarkdown is described as being small, so perhaps that's a good trait to consider.

**Comment by ArnoldSmith86** [comment]
> Line break should work with `\n` in the JSON? I think?

At least on card objects they don't. I'm pretty sure it's the same for labels.

**Comment by robartsd** [comment]
I like the idea of markdown support for both labels and card text.

**Comment by robartsd** [comment]
> > Line break should work with `\n` in the JSON? I think?
> 
> At least on card objects they don't. I'm pretty sure it's the same for labels.

They will soon. PR approved, merge pending 24 hour waiting period.

**Comment by bjalder26** [comment]
Would it work for labels to be made from contenteditable div instead of from textareas?  Then I'm thinking it would be possible for people to use html tags, and if it works, then I would think there would only be a relatively little JS to add in order to give in-line changes.

Contenteditable div example: https://jsfiddle.net/plantface/8zevmL4d/
A project to make contenteditable div behave more like textareas (including getting text only - not tags): https://jsfiddle.net/UziTech/4msdgjox/

**Comment by bjalder26** [comment]
I've been fiddling around with this idea, and I've had a couple of thoughts.
1) I think It would be possible to allow game makers to utilize html tags in 'labels', however the 'label' (presumably a different on, or one with a specific property) would have to have the html inbetween the div tags, rather than a textarea.  The image below was produced by pasting in html in a div. 

I presume this approach would require a textarea to accept the html tags, but then the html tags would need to replace the text area, or rather they would need to be able to be toggled back and forth.

2)  Another approach could be to allow a simple markdown such as **text** for bold and *text* for italics.  Then have that converted to html using designer-defined replace methods  ( e.g. .replace(/\*\*(.*)\*\*/gim, '<b>$1</b>') ) that were stored in a label with a specific id (e.g. "markdown").  That way they could copy and paste the replacement they would like to have and it could be even easier. The html for the image below was also made using this method from a little javascript (https://jsfiddle.net/bjalder26/c3xdj4hb/15/ - I separated the replacements into a second function).

![image](https://user-images.githubusercontent.com/78324099/121797697-7d007d00-cbe7-11eb-8f81-e67969f83a62.png)

The point of doing it one of these two ways (if you want this at all), would be that it would give a lot of flexibility, not require much javascript, and not require more features to be added later.

If this is something that is actually desired, then I might be able to work it out, or work it out with some help.  However, if it is not, then that is fine by me.  I'm pretty sure I could figure out how to do anything I wanted to do within the JSON itself.

**Comment by bjalder26** [comment]
#582 is related

**Comment by RaphaelAlvez** [comment]
@bjalder26 I have now merged the basic UI. The next step was to add css properties to labels and add input for those in the label editor.
This PR clearly make that obsolete. Do you think you could include the same markdown in a text area in the label editor?
It would be nice if it had buttons for each markdown. If you created one I could add the others by copying the code.
